src/test/java/org/xwiki/git/script/GitScriptServiceTest.java

```diff
@@ -114,4 +117,35 @@ public void getCountCommits() throws Exception
         assertEquals("test author", commits[0].getName());
         assertEquals("author@doe.com", commits[0].getEmail());
     }
+
+    @Test
+    public void getRepositoryBareAndCountCommits() throws Exception
+    {
+        GitScriptService service = this.componentManager.getInstance(ScriptService.class, "git");
+        Repository repository = service.getRepositoryBare(this.testRepository.getAbsolutePath(), TEST_REPO_CLONED);
+        try {
+            assertNull(repository.getWorkTree());
+            fail("Expected NoWorkTreeException");
+        } catch (NoWorkTreeException e) {
+            UserCommitActivity[] commits = service.countAuthorCommits(1, repository);
+            assertEquals(1, commits.length);
+            assertEquals(1, commits[0].getCount());
+        }
+    }
+
+    @Test
+    public void getRepositoryBareWithCredentialsAndFindAuthors() throws Exception
+    {
+        GitScriptService service = this.componentManager.getInstance(ScriptService.class, "git");
+        Repository repository = service.getRepositoryBare(this.testRepository.getAbsolutePath(), TEST_REPO_CLONED,
+            "test author", "TestAccessCode");
+        try {
+            assertNull(repository.getWorkTree());
+            fail("Expected NoWorkTreeException");
+        } catch (NoWorkTreeException e) {
+            Set<PersonIdent> authors = service.findAuthors(repository);
+            assertEquals(1, authors.size());
+            assertEquals("test author", authors.iterator().next().getName());
+        }
```

Comment:
I have the feeling there's a lot of duplication in tests here over the existing tests.

Reply:
Minimized the duplication now.

Reply:
@Haxsen I think the duplication of use cases exists in your update implementation.
For example, the use cases of counting commits and finding authors are being tested twice both for bare repository and a normal one.
Just a single test that simply tests if a repository is bare should be enough for testing bare repository and covering the code. The other use cases are already covered. Or better, you can test a use case unique to a bare repository.

Reply:
@9inpachi  I see, if that is what it means then let me update this PR in an hour.
Thanks a lot.

Reply:
Hi @9inpachi , I have fixed this, can you confirm? Thanks.

Reply:
Great @Haxsen 
Looks much simpler now.